tooling/metamodel-generator/src/main/java/org/hibernate/processor/annotation/AnnotationMetaEntity.java

```diff
@@ -2452,7 +2453,7 @@ private void createSingleParameterFinder(
 							new CriteriaFinderMethod(
 									this, method,
 									methodName,
-									returnType.toString(),
+									typeAsString( returnType, false ),
```

Comment:
@marko-bekhta there are three places in `AnnotationMetaEntity` where this constructor is called. I believe this change should be replicated to all three invocations, do you agree?